src/SerializeSourcemap.cc

```diff
@@ -344,29 +344,98 @@ sos::Object WrapResourceSourcemap(const SourceMap<Resource>& resource)
     return resourceObject;
 }
 
-sos::Object WrapResourceGroupSourcemap(const SourceMap<ResourceGroup>& resourceGroup)
+sos::Object WrapResourceGroupSourcemap(const SourceMap<Element>& resourceGroup)
 {
     sos::Object resourceGroupObject;
 
     // Name
-    resourceGroupObject.set(SerializeKey::Name, WrapSourcemap(resourceGroup.name));
+    resourceGroupObject.set(SerializeKey::Name, WrapSourcemap(resourceGroup.attributes.name));
 
-    // Description
-    resourceGroupObject.set(SerializeKey::Description, WrapSourcemap(resourceGroup.description));
-
-    // Resources
+    // Description & Resources
+    SourceMap<Description> description;
     sos::Array resources;
 
-    for (Collection<SourceMap<Resource> >::const_iterator it = resourceGroup.resources.collection.begin();
-         it != resourceGroup.resources.collection.end();
+    for (Collection<SourceMap<Element> >::const_iterator it = resourceGroup.content.elements().collection.begin();
+         it != resourceGroup.content.elements().collection.end();
          ++it) {
 
-        resources.push(WrapResourceSourcemap(*it));
+        if (it->element == Element::ResourceElement) {
```

Comment:
This possibly indicates some code smell – elements should be either 1. wrapped in the same way or 2. overloaded / templated WrapElement() should take care of it